tests/emex_upload/test_services.py

```
@@ -0,0 +1,19 @@
+# TODO: Rewrite tests for the new streaming validation service (`validate_file_and_animate_progress`).
+# The previous tests for the legacy `process_emex_file` function were removed.
+# Those tests covered the following validation scenarios, which should be re-implemented:
+#
+# 1.  **File Structure:**
+#     - A file with the wrong number of columns should be processed, and the malformed rows should be counted as structural errors.
+#
+# 2.  **Data Integrity:**
+#     - Rows with non-numeric text in numeric columns (e.g., "нет цены" in "ЦенаПокупки") should be flagged and counted.
+#
+# 3.  **Business Logic:**
+#     - Rows with invalid values in the 'Склад' column (i.e., not 'ДА' or 'НЕТ') should be flagged.
```

Ruff (0.13.3)

Comment contains ambiguous `Н` (CYRILLIC CAPITAL LETTER EN). Did you mean `H` (LATIN CAPITAL LETTER H)?

(RUF003)